lib/request.js

```diff
@@ -0,0 +1,102 @@
+'use strict'
+const http = require('http')
+const https = require('https')
+const http2 = require('http2')
+const eos = require('end-of-stream')
+const pump = require('pump')
+const { stripHttp1ConnectionHeaders } = require('./utils')
+
+class RequestAgent {
+  constructor (opts) {
+    this.http2 = !!opts.http2
+    this.requests = {
+      'http:': http,
+      'https:': https
+    }
+    if (this.http2) {
+      if (!opts.base) throw new Error('Option base is required when http2 is true')
+      this.http2Client = http2.connect(opts.base, { rejectUnauthorized: opts.rejectUnauthorized })
+    } else {
+      this.agents = {
+        'http:': new http.Agent(agentOption(opts)),
+        'https:': new https.Agent(agentOption(opts))
+      }
+    }
+  }
+
+  handleHttp1Req ({ method, url, qs, headers, body }, done) {
```

Comment:
The deconstructing assignment is slower than the classic `obj.key` assignment. Can you use it?